src/main/java/com/arpnetworking/metrics/impl/ApacheHttpSink.java

```diff
@@ -312,32 +312,27 @@ private void dispatchRequest(final CloseableHttpClient httpClient, final ClientV
                 }
             }
 
-            //TODO(brandonarp): just pull from dimensions once the library supports it
-            final String host = event.getAnnotations().get("_host");
-            if (host != null) {
-                builder.addDimensions(
-                        ClientV3.DimensionEntry.newBuilder()
-                                .setName(createIdentifier("host"))
-                                .setValue(createIdentifier(host))
-                                .build());
-            }
-
-            final String service = event.getAnnotations().get("_service");
-            if (service != null) {
-                builder.addDimensions(
-                        ClientV3.DimensionEntry.newBuilder()
-                                .setName(createIdentifier("service"))
-                                .setValue(createIdentifier(service))
-                                .build());
-            }
+            for (final Map.Entry<String, String> dimension : event.getAnnotations().entrySet()) {
+                final String key = dimension.getKey();
+                final String value = dimension.getValue();
+                if ("_host".equals(key) || "_service".equals(key) || "_cluster".equals(key)) {
+                    // Special prescribed dimensions
+                    builder.addDimensions(
+                            ClientV3.DimensionEntry.newBuilder()
+                                    .setName(createIdentifier(key.substring(1)))
+                                    .setValue(createIdentifier(value))
+                                    .build());
+                } else if (!"_end".equals(key) && !"_start".equals(key) && !"_id".equals(key)) {
+                    // User specified dimensions
+                    builder.addDimensions(
+                            ClientV3.DimensionEntry.newBuilder()
+                                    .setName(createIdentifier(key))
+                                    .setValue(createIdentifier(value))
+                                    .build());
+                }
+                // "_end", "_start" and "_id" are reserved annotation keys which
```

Comment:
Instead of the comment, I think this would be clearer if you had a static set
```java
private static final Set<String> reservedAnnotationKeys =
    ImmutableSet.of("_id", "_start", "_end");
//
// ...
//
} else if (!reservedAnnotationKeys.contains(key) {
```

Reply:
Done!

Reply:
🤘